File: swea/D2/1928.Base64 Decoder/1928.cpp
#include <iostream>
#include <string>
#include <bitset>

using namespace std;

string table = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";

int T;

int main(){
  cin >> T;

  for(int t = 1; t <= T; t++){
    string code;
    cin >> code;

    //글자마다 그에 대응하는 6자리 2진수로 변환 후, 모아두기
    string decode = "";
    for(int i = 0; i < code.length(); i++){
      bitset<6> b(table.find(code[i]));
      decode += b.to_string();
    }
    
    //8자리씩 잘라서 해독하기.
    string ans = "";
    for(int i = 0; i < decode.length(); i+=8){
      string tmp = decode.substr(i, 8);
      int n = stoi(tmp, nullptr, 2);
      ans += (char)n;
    }

    cout << '#' << t << ' ' << ans << endl;
  }
}

File: swea/D2/1928.Base64 Decoder/1928.py
base_table ='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'

T = int(input())
for tc in range(1, T+1):
    s = input()

    binary = ""
    for i in s:
        #테이블에 저장된 문자의 인덱스를 구하고, 그것을 6자리 이진수로 변환
        binary += format(base_table.find(i), '06b')

    ans = ""
    for k in range(0, len(binary), 8):
        #8개씩 잘라서 만든 8자리 이진수를 10진법으로 변환한 후, 문자로 변환
        ans += chr(int(binary[k:k+8], 2))

    print(f"#{tc} {ans}")